server/server.c

```diff
@@ -0,0 +1,111 @@
+#include "server.h"
+
+void socketThread(int clientSocket)
+{
+    char client_message[2000];
+    char buffer[1024];
+    int newSocket = clientSocket;
+
+    //Receive a reply from the client
+    if (recv(newSocket, client_message, 2000, 0) < 0)
+    {
+        perror("Recv failed");
+        exit(1);
+    }
+
+    // Send message to the client socket
+    pthread_mutex_lock(&lock);
+    print_directory_tree(".");
+
+    char *message = malloc(sizeof(client_message) + 20);
+    strcpy(message, "Hello Client : ");
+    strcat(message, client_message);
+    strcat(message, "\n");
+    strcpy(buffer, message);
+    free(message);
+    pthread_mutex_unlock(&lock);
+    sleep(1);
+    send(newSocket, buffer, 13, 0);
+    printf("Exit socketThread \n");
+    close(newSocket);
+}
+
+int main()
+{
+    short serverPort = 8080;
```

Comment:
The port and IP should be common for the client and server.

Reply:
Done